src/util/BitField.js

```diff
@@ -38,10 +38,11 @@ class BitField {
   /**
    * Checks whether the bitfield has a bit, or multiple bits.
    * @param {BitFieldResolvable} bit Bit(s) to check for
+   * @param {...*} hasParams Additional parameters for the has method, if any
```

Comment:
I'm not sure what exactly this solves, but I'd rather drop it and the recursion that requires it.
`BitField.resolve` handles arrays just fine.

Reply:
Without this, it used to return true whenever they had admin no matter if you actually passed true to checkAdmin.
I still do not really understand the issue here, anyway as you said `<BitField>.constructor.resolve` handles arrays well already.